cpg-concepts/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/concepts/file/File.kt

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2024, Fraunhofer AISEC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *                    $$$$$$\  $$$$$$$\   $$$$$$\
+ *                   $$  __$$\ $$  __$$\ $$  __$$\
+ *                   $$ /  \__|$$ |  $$ |$$ /  \__|
+ *                   $$ |      $$$$$$$  |$$ |$$$$\
+ *                   $$ |      $$  ____/ $$ |\_$$ |
+ *                   $$ |  $$\ $$ |      $$ |  $$ |
+ *                   \$$$$$   |$$ |      \$$$$$   |
+ *                    \______/ \__|       \______/
+ *
+ */
+package de.fraunhofer.aisec.cpg.graph.concepts.file
+
+import de.fraunhofer.aisec.cpg.graph.Node
+import de.fraunhofer.aisec.cpg.graph.concepts.Concept
+import de.fraunhofer.aisec.cpg.graph.concepts.Operation
+
+/**
+ * This interface indicates that the corresponding node is connected to a file concept or operation.
+ */
+interface IsFile
+
+/**
+ * Represents a file access mode flag.
+ *
+ * man 0p fcntl.h
+ *
+ * ```
+ * The <fcntl.h> header shall define the following symbolic constants for use as the  file access  modes  for  open(),  openat(), and fcntl().  The values shall be unique, except that O_EXEC and O_SEARCH may have equal values. The values shall be suitable for use in #if preprocessing directives.
+ *
+ * O_EXEC      Open for execute only (non-directory files). The result is  unspecified  if this flag is applied to a directory.
+ *
+ * O_RDONLY    Open for reading only.
+ *
+ * O_RDWR      Open for reading and writing.
+ *
+ * O_SEARCH    Open  directory  for search only. The result is unspecified if this flag is applied to a non-directory file.
+ *
+ * O_WRONLY    Open for writing only.
+ * ```
+ */
+enum class FileAccessModeFlags(val value: Long) : IsFile {
```

Comment:
These names (while being those of Linux) are somehow hard to read. Why are some not supported? Individual entries lack documentation. Can we somehow easily query if a file was opened/is in state r, w, rw, x, ... or do we have to match on these names and/or parse the access mode masks manually? A utility function doing this would be really useful